istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,714 @@
+			opts.webhookSecretNameSpace = namespace
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating Kubernetes client interface: %v", err)
+			}
+			err = enableWebhookConfig(client, opts)
+			if err != nil {
+				return fmt.Errorf("err when enabling webhook configurations: %v", err)
+			}
+			fmt.Println("webhook configurations have been enabled")
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.enableValidationWebhook, "validation", true, "Enable "+
+		"validatation webhook (default true).")
+	flags.BoolVar(&opts.enableMutationWebhook, "injection", true, "Enable "+
+		"injection webhook (default true).")
+	// Specifies the local file path to webhook CA bundle.
+	// If empty, will read CA bundle from default Kubernetes service account.
+	flags.StringVar(&opts.caCertPath, "ca-bundle-file", "",
+		"PEM encoded CA bundle which will be used to validate the webhook's server certificates. "+
+			"If this is empty, the kube-apisever's root CA is used if it can be confirmed to have signed "+
+			"the webhook's certificates. This condition is sometimes true but is not guaranteed "+
+			"(see https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping)")
+	flags.StringVar(&opts.validationWebhookConfigPath, "validation-path", "",
+		"The file path of the validation webhook configuration.")
+	flags.StringVar(&opts.mutatingWebhookConfigPath, "injection-path", "",
+		"The file path of the injection webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookServiceName, "validation-service", "istio-galley",
+		"The service name of the validation webhook to manage.")
+	flags.StringVar(&opts.mutatingWebhookServiceName, "injection-service", "istio-sidecar-injector",
+		"The service name of the injection webhook to manage.")
+	flags.StringVar(&opts.webhookSecretName, "webhook-secret", "",
+		"The name of the secret of a webhook. istioctl will verify that the webhook certificate "+
+			"is issued by the CA certificate.")
+	flags.DurationVar(&opts.maxTimeForCheckingWebhookServer, "timeout", 60*time.Second,
+		"	Max time for checking the validating webhook server. If the validating webhook server is not ready"+
+			"in the given time, exit. Otherwise, apply the webhook configuration.")
+	flags.DurationVar(&opts.maxTimeForReadingWebhookCert, "read-cert-timeout", 60*time.Second,
+		"	Max time for waiting the webhook certificate to be readable.")
+
+	return cmd
+}
+
+func newDisableCmd() *cobra.Command {
+	opts := &disableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "disable",
+		Short: "Disable webhook configurations",
+		Example: `
+# Disable all webhooks
+istioctl experimental post-install webhook disable
+
+# Disable all webhooks except injection
+istioctl experimental post-install webhook disable --injection=false
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Do we want to add a `--yes` or interactive "Are you sure (?)" the way we do in _istio.io/operator/cmd/mesh/manifest-apply.go_?  Just trying this out I blew away my webhook and don't know how to recreate it.  I am installing Istio 1.3.2 because I don't know how to get back to a good 1.3.1.

Reply:
Done.